src/melee.cpp

```diff
@@ -1340,6 +1340,30 @@ bool player::can_weapon_block() const
             weapon.has_technique( WBLOCK_3 ));
 }
 
+int blocking_ability( const item &shield )
+{
+    int block_bonus = 2;
+    if (shield.has_technique( WBLOCK_3 )) {
+        block_bonus = 10;
+    } else if (shield.has_technique( WBLOCK_2 )) {
+        block_bonus = 6;
+    } else if (shield.has_technique( WBLOCK_1 )) {
+        block_bonus = 4;
+    }
+    return block_bonus;
+}
+
+item &player::best_shield()
+{
+    int weapon_block = blocking_ability( weapon );
+    for( const item &shield : worn ) {
+        if( shield.has_flag( "BLOCK_WHILE_WORN" ) && blocking_ability( shield ) >= weapon_block ) {
+            return shield;
+        }
+    }
+    return weapon;
```

Comment:
Changing this one line to an `if` could make the later sections much easier:
```
if( can_weapon_block() ) {
    return weapon;
}

return ret_null;
```

This way if it returns an item that is not null, we know it returned an item we can actually use to block stuff with.